superset/connectors/sqla/models.py

```diff
@@ -2216,6 +2216,11 @@ def write_shadow_dataset(  # pylint: disable=too-many-locals
         engine = database.get_sqla_engine(schema=dataset.schema)
         conditional_quote = engine.dialect.identifier_preparer.quote
 
+        new_dataset = session.query(NewDataset).one_or_none()
```

Comment:
I'm confused about this... this will return **any** (new) dataset that exists. We should filter by a key here.

Reply:
Whew, thanks for catching that.